### ISSUE TITLE
`@llamaindex/env` TypeError: Module "..." needs an import attribute of "type: json"

### ISSUE THREAD

**Issue by parhammmm** [run-llama/LlamaIndexTS#957]
When trying to use the latest `llamaindex` 0.4.1 I get the following compilation error when running tests in `vitest`

This is likely not a LITS issue, but just wonder if anyone knows the reason I can't figure this out!

```bash

TypeError: Module "file:///Users/p/LlamaIndexTS/node_modules/.pnpm/tiktoken@1.0.15/node_modules/tiktoken/encoders/cl100k_base.json" needs an import attribute of "type: json"

```

**Comment by travellingprog** [comment]
This seems to be the problematic line, which was added last week:
https://github.com/run-llama/LlamaIndexTS/blob/main/packages/env/src/tokenizers/node.ts#L2

**Comment by marcusschiesser** [comment]
I had to add `resolveJsonModule: true` to `tsconfig.json` of `@llamaindex/env` for this to work:

```json
{
  "compilerOptions": {
    "resolveJsonModule": true
  },
}
```

my guess is as we're not bundling, you have to use this setting for any project (what we don't want). 

@travellingprog I tried your idea, it only works with module set to `esnext` 

@himself65 any idea?

**Comment by himself65** [comment]
> I had to add `resolveJsonModule: true` to `tsconfig.json` of `@llamaindex/env` for this to work:
> 
> ```json
> {
>   "compilerOptions": {
>     "resolveJsonModule": true
>   },
> }
> ```
> 
> my guess is as we're not bundling, you have to use this setting for any project (what we don't want).
> 
> @travellingprog I tried your idea, it only works with module set to `esnext`
> 
> @himself65 any idea?

It's only for TS compilation time, not user-side runtime.

let me fix it

**Comment by parhammmm** [comment]
Perfeft thanks everyone!

**Comment by parhammmm** [comment]
@himself65 seems like the build failed https://github.com/run-llama/LlamaIndexTS/actions/runs/9618158421/job/26531444358

Another solution here could be that the json is read as a file instead of importing it, what do you think?

**Comment by himself65** [comment]
> @himself65 seems like the build failed [run-llama/LlamaIndexTS/actions/runs/9618158421/job/26531444358](https://github.com/run-llama/LlamaIndexTS/actions/runs/9618158421/job/26531444358)
> 
> Maybe the file needs to included in the bundle?

It seems like JSR issue

**Comment by marcusschiesser** [comment]
@parhammmm does https://github.com/run-llama/LlamaIndexTS/pull/958 fix your issue? If not how can we reproduce it?

I am running into the same issue with the express template from create-llama. I tried 0.4.1 (in this PR https://github.com/run-llama/create-llama/pull/143) and the latest main branch (containing https://github.com/run-llama/LlamaIndexTS/pull/958)  locally (using `pnpm add`) - both ways the result is:

```
pnpm run start

> my-app@0.1.0 start /Users/marcus/1/my-app
> node dist/index.js

node:internal/modules/esm/assert:89
        throw new ERR_IMPORT_ASSERTION_TYPE_MISSING(url, validType);
              ^

TypeError [ERR_IMPORT_ASSERTION_TYPE_MISSING]: Module "file:///Users/marcus/code/llamaindex/LlamaIndexTS/node_modules/.pnpm/tiktoken@1.0.15/node_modules/tiktoken/encoders/cl100k_base.json" needs an import attribute of type "json"
    at validateAttributes (node:internal/modules/esm/assert:89:15)
    at defaultLoad (node:internal/modules/esm/load:153:3)
    at async ModuleLoader.load (node:internal/modules/esm/loader:403:7)
    at async ModuleLoader.moduleProvider (node:internal/modules/esm/loader:285:45) {
  code: 'ERR_IMPORT_ASSERTION_TYPE_MISSING'
}
```

**Comment by marcusschiesser** [comment]
@parhammmm https://github.com/run-llama/LlamaIndexTS/pull/967 is fixing the issue for the create-llama express template

**Comment by himself65** [comment]
> @parhammmm #967 is fixing the issue for the create-llama express template

could you add a e2e test in this packages/llamaindex/e2e/...?

**Comment by himself65** [comment]
> > @parhammmm #967 is fixing the issue for the create-llama express template
> 
> could you add a e2e test in this packages/llamaindex/e2e/...? when you find a way to fix such issue

https://github.com/run-llama/LlamaIndexTS/pull/963

**Comment by parhammmm** [comment]
@marcusschiesser was checking but https://github.com/run-llama/LlamaIndexTS/pull/978 got in the way

**Comment by marcusschiesser** [comment]
@parhammmm thanks, just released 0.4.5 containing it

**Comment by parhammmm** [comment]
@marcusschiesser looks like this is resolved! thanks again
